azure/services/managedclusters/managedclusters.go

```diff
@@ -240,6 +241,10 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		},
 	}
 
+	if tags := *to.StringMapPtr(managedClusterSpec.Tags); len(tags) != 0 {
```

Comment:
Can you explain why this change (only add tags property value if it is present in ManagedCluster spec) allows tags to be updated?

Reply:
No it's because when tags are empty the existing object returned from the azure api has a nil value for tags but the current object has map[string]string with length 0, the diff function we use is returning true for this case, which triggers unnecessary updates.